src/commands/deploy/deploy.ts

```diff
@@ -59,8 +59,11 @@ export async function deploy(actionContext: IActionContext, arg1?: vscode.Uri |
     }
 
     const isZipDeploy: boolean = siteConfig.scmType !== constants.ScmType.LocalGit && siteConfig !== constants.ScmType.GitHub;
+    const appSettingsClient: IAppSettingsClient = node.client;
+    const remoteSettings: WebSiteManagementModels.StringDictionary = await appSettingsClient.listApplicationSettings();
+    const isBuildDuringDeploymentAppSetting: boolean = !!remoteSettings.properties && 'SCM_DO_BUILD_DURING_DEPLOYMENT' in remoteSettings.properties;
```

Comment:
How about "isBuildDuringDeploymentAppSetting" -> "hasScmDoBuildSetting"

"has" makes more sense than "is" and I think getting the first part of "SCM_DO_BUILD_DURING_DEPLOYMENT" in the name is more important than "DuringDeployment"